x-pack/modules/azure/lib/filters/azure_event.rb

```diff
@@ -0,0 +1,246 @@
+            group = event.get("[azure][group]")
+            category = event.get("[azure][category]")
+            explode_properties(group, category, event)
+            event.set("[#{group}_#{category}_properties]", event.get("properties"))
+            event.remove("properties")
+          end
+          filter_matched(event)
+        else
+          @tag_on_failure.each do |tag|
+            event.tag(tag)
+          end
+          logger.trace("Failed to parse event as an Azure Event", :event => event) if logger.trace?
+        end
+      end
+
+      # public methods may change state of event, and should return true if successful
+
+      ##
+      # Parses the resourceId to pull out the relevant parts into the event
+      # return true if the common set of fields can be extracted from the resourceId, false otherwise
+      def add_resources(resource_id, event)
+        common_resources = explode_common_resources(resource_id)
+        if common_resources
+          common_resources.each do |k, v|
+            event.set("[azure][#{k}]", v)
+          end
+
+          if common_resources["provider"] == "MICROSOFT.SQL"
+            sql_resources = explode_sql_resources(resource_id)
+            if sql_resources
+              sql_resources.each do |k, v|
+                event.set("[azure][#{k}]", v)
+              end
+            end
+          end
+        end
+        !common_resources.nil?
+      end
+
+      ##
+      # adds a group and category to the provided event
+      # return true if the group and category are added, false otherwise
+      def add_group_and_category(event)
+        #always prefer @metadata category over the heuristics
+        category = event.get("[@metadata][category]")
+        if category.nil?
+          category = find_category(event)
+        end
+        event.set("[azure][category]", category) if category
+        group = @activity_log_categories.include?(category) ? ACTIVITY_LOG_GROUP : @sql_diagnostics_categories.include?(category) ? SQL_DIAGNOSTICS_GROUP : nil
+        event.set("[azure][group]", group) if group
+
+        !category.nil?
+      end
+
+      ##
+      # Based on the group and category, explode the custom properties bag
+      def explode_properties(group, category, event)
+        if group == ACTIVITY_LOG_GROUP && category == "ServiceHealth"
+          add_json_event_properties(event)
+        end
+      end
+
+      # Some events such as Service Health can have a Json inside Json as [properties][eventProperties]. This will explode the inner json and replace it with indivisual fields on the event.
+      def add_json_event_properties(event)
+        inner_json = event.get("[properties][eventProperties]")
+        inner_json_as_hash = json_to_hash(inner_json)
+        if inner_json_as_hash
+          inner_json_as_hash.each do |k, v|
+            event.set("[properties][#{k}]", v)
+          end
+          event.remove("[properties][eventProperties]")
```

Comment:
could there be an `eventProperties` json field in the `[properties][eventProperties]` json object that would be removed in this line? we can reorder the operations here and remove first then set

Reply:
This only gets removed if the `[properties][eventProperties]` is indeed json itself. the `json_to_hash` method will return nil if it is not json and won't remove unless it actually json.